docs/components/integration-framework/connectors/custom-built-connectors/connector-sdk.md

```diff
@@ -4,17 +4,677 @@ title: Connector SDK
+
+    // (2)
+    context.validate(connectorRequest);
+    // (3)
+    context.replaceSecrets(connectorRequest);
+
+    return executeConnector(connectorRequest);
+  }
+
+  private MyConnectorResult executeConnector(final MyConnectorRequest connectorRequest) {
+    LOGGER.info("Executing my connector with request {}", connectorRequest);
+    var result = new MyConnectorResult();
+    // (4)
+    result.setMyProperty("Message received: " + connectorRequest.getMessage());
+    return result;
+  }
+}
+```
+
+The `execute` method receives all necessary environment data via the `ConnectorContext` object.
+The connector runtime environment initializes the context. It allows to
+
+- Fetch and deserialize the input data as shown in **(1)**. See [input data](#input-data) section for details.
+- Validate the created request object as shown in **(2)**. See [validation](#validation) section for details.
+- Replace secrets in the request object as shown in **(3)**. See [secrets](#secrets) section for details.
+
+If the connector has a result to return, it can create a new result data object and set
+its properties as shown in **(4)**.
+
+Using this outline, you start the business logic of your connector in the `executeConnector` method
+and expand from there.
+
+#### Input data
+
+The input data of a connector is provided by the process instance that executes the connector.
+You can either fetch this data as a raw JSON String using the context's `getVariables` method
+or deserialize the data into your own request object directly with the `getVariablesAsType`
+method shown in **(1)**.
+
+Using `getVariablesAsType` will attempt to deserialize the JSON String containing the input
+data into Java objects. This deserialization depends on the connector runtime environment your
+connector function runs in. Thus, use this deserialization approach with caution.
+While it works reliably for many input data types like String, Boolean, Integer, and nested
+objects, you might want to consider deserializing your connector's input data in a custom fashion
+using `getVariables` and a library like [Gson](https://github.com/google/gson).
+
+The `getVariablesAsType` method and tools like Gson can properly reflect nested data
+objects. You can define nested structures by referencing other Java classes as attributes.
+Looking at the `authentication` data input example described in the [connector template](#connector-template),
+you can create the following input data objects to reflect the structure properly:
+
+```java
+package io.camunda.connector;
+
+public class MyConnectorRequest {
+
+  private String message;
+  private Authentication authentication;
+}
+```
+
+```java
+package io.camunda.connector;
+
+public class Authentication {
+
+  private String user;
+  private String token;
+}
+```
+
+#### Validation
+
+Validating input data is a common task in a connector function. Therefore, the SDK provides an
```

Comment:
Minor tweak:
Update text from "Therefore, the SDK provides an..."
to "The SDK provides an..."

Reply:
Updated